# Lab: Building a Reproducible Analysis Workflow

Choose your own documented dataset, develop three distinct research questions, and answer each using the six-step workflow from the Pokémon lecture. You may reuse your visual thinking dataset, but develop new questions rather than repeat its single-feature charts.


## The reproducible analysis workflow

1. **Question:** What do we want to learn?
2. **Data:** Which observations and features can help answer it?
3. **Evidence:** What should we calculate and display to answer the question?
4. **Check:** Does the result make sense in context? Do a few original records support the meaning of the calculation?
5. **Conclusion:** What claim is supported?
6. **Limitation:** What should we avoid concluding?


## Learning goals

- Develop three questions that can be answered with your chosen data.
- Inspect and document a dataset using its data dictionary.
- Select, group, summarize, or calculate features with Pandas.
- Connect displayed evidence, contextual sanity checks, conclusions, and limitations.


## Using AI tools responsibly

You may use AI to ask questions, debug code, or receive feedback. You remain responsible for understanding the work and checking suggestions. Disclose whether you used AI in the final reflection.


## Find and document your dataset

Choose your own documented CSV dataset from [Kaggle](https://www.kaggle.com/datasets) or another reputable source. You may reuse the dataset from the visual thinking lab. Choose data with:

- a clearly identified publisher or creator;
- a data dictionary or documentation explaining observations and features;
- at least one categorical and one quantitative feature;
- enough information to answer three distinct questions involving calculations or comparisons;
- content appropriate to share in class, without private or personally identifying information; and
- a manageable file size.

Save the CSV in this notebook's `data` folder. Use a descriptive DataFrame name ending in `_df` throughout your analysis.


**What is the dataset's title, publisher or creator, direct source URL, and data dictionary URL or location?**

Your answer:


**Briefly describe the dataset and relevant features using its documentation. What does one row represent? Include units and the meaning of blank values where documented.**

Your answer:


Dataset Title: Video Games Sales Dataset
Publisher/Creator: SID_TWR
Direct Source URL: https://www.kaggle.com/datasets/sidtwr/videogames-sales-dataset
Data Dictionary/Documentation: The data dictionary is provided in the "About Dataset"/metadata section of the Kaggle dataset page, where the dataset's features and their meanings are documented.

## Plan three investigations

Develop three distinct research questions about your dataset. Each must require a calculation or comparison, not simply looking up a value. Consider grouped averages, percentages meeting a condition, or comparisons between features, as in the lecture; adapt the questions to what your data means. Complete all six steps for each question. Record each prediction before running its calculations.


## Investigation 1

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 1?**

Your answer:


Which video game genre has the highest average global sales?

### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer:


I predict that the Shooter genre will have the highest average global sales because many widely popular video game franchises are shooters and are released across major gaming platforms.

### Step 2: Data


Import Pandas and load your CSV into a descriptively named DataFrame ending in `_df`. Inspect the data once here, using separate cells for each operation.


In [1]:
import pandas as pd


In [3]:
video_games_df = pd.read_csv("data/Video_Games_Sales_as_at_22_Dec_2016.csv")


In [4]:
video_games_df.head()


,Name,Platform,Year_of_Release,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales,Critic_Score,Critic_Count,User_Score,User_Count,Developer,Rating
0,Wii Sports,Wii,2006.0,Sports,Nintendo,41.36,28.96,3.77,8.45,82.53,76.0,51.0,8.0,322.0,Nintendo,E
1,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24,NaN,NaN,NaN,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.68,12.76,3.79,3.29,35.52,82.0,73.0,8.3,709.0,Nintendo,E
3,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.61,10.93,3.28,2.95,32.77,80.0,73.0,8.0,192.0,Nintendo,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37,NaN,NaN,NaN,NaN,NaN,NaN


In [5]:
video_games_df.tail()


,Name,Platform,Year_of_Release,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales,Critic_Score,Critic_Count,User_Score,User_Count,Developer,Rating
16714,Samurai Warriors: Sanada Maru,PS3,2016.0,Action,Tecmo Koei,0.00,0.00,0.01,0.0,0.01,NaN,NaN,NaN,NaN,NaN,NaN
16715,LMA Manager 2007,X360,2006.0,Sports,Codemasters,0.00,0.01,0.00,0.0,0.01,NaN,NaN,NaN,NaN,NaN,NaN
16716,Haitaka no Psychedelica,PSV,2016.0,Adventure,Idea Factory,0.00,0.00,0.01,0.0,0.01,NaN,NaN,NaN,NaN,NaN,NaN
16717,Spirits & Spells,GBA,2003.0,Platform,Wanadoo,0.01,0.00,0.00,0.0,0.01,NaN,NaN,NaN,NaN,NaN,NaN
16718,Winning Post 8 2016,PSV,2016.0,Simulation,Tecmo Koei,0.00,0.00,0.01,0.0,0.01,NaN,NaN,NaN,NaN,NaN,NaN


In [6]:
video_games_df.shape


(16719, 16)

In [7]:
video_games_df.info()


<class 'pandas.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  str    
 1   Platform         16719 non-null  str    
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  str    
 4   Publisher        16665 non-null  str    
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       7590 non-null   float64
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  str    
 15  Rating           9950 non-null   str    
dtypes: float64(10), str(6)
memory usage: 2.0 MB


In [8]:
video_games_df.isnull().sum()


Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         9129
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer:


I will use the Genre and Global_Sales features to compare average global video game sales across genres. Each observation represents a video game release on a particular platform. Genre is the categorical feature used to group the games, and Global_Sales is the quantitative feature measured in millions of units sold. I will exclude observations with missing Genre or Global_Sales values from the calculation because they cannot be used to determine a genre-specific average. Since this investigation compares averages rather than percentages, a denominator does not need to be defined.

### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [9]:
genre_sales_df = video_games_df.dropna(subset=["Genre", "Global_Sales"])
genre_sales_df[["Genre", "Global_Sales"]].head()


,Genre,Global_Sales
0,Sports,82.53
1,Platform,40.24
2,Racing,35.52
3,Sports,32.77
4,Role-Playing,31.37


In [10]:
genre_sales_df.groupby("Genre")["Global_Sales"].mean().sort_values(ascending=False)


Genre
Platform        0.932523
Shooter         0.795873
Role-Playing    0.622933
Racing          0.583587
Sports          0.567291
Fighting        0.527067
Action          0.517884
Misc            0.458960
Simulation      0.446705
Puzzle          0.419000
Strategy        0.255490
Adventure       0.182417
Name: Global_Sales, dtype: float64

**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer:


The Platform genre has the highest average global sales at approximately 0.93 million units per game. Shooter games have the second-highest average at approximately 0.80 million units, followed by Role-Playing games at approximately 0.62 million units. I calculated these values by grouping the observations by Genre and calculating the mean Global_Sales for each group. I then sorted the averages from highest to lowest, which showed that Platform games have the highest average global sales.

### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [11]:
genre_sales_df.groupby("Genre")["Global_Sales"].agg(["count", "min", "max", "mean"]).sort_values("mean", ascending=False)


,count,min,max,mean
Genre,,,,
Platform,888,0.01,40.24,0.932523
Shooter,1323,0.01,28.31,0.795873
Role-Playing,1500,0.01,31.37,0.622933
Racing,1249,0.01,35.52,0.583587
Sports,2348,0.01,82.53,0.567291
Fighting,849,0.01,12.84,0.527067
Action,3370,0.01,21.04,0.517884
Misc,1750,0.01,28.92,0.458960
Simulation,874,0.01,24.67,0.446705


**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer:


I would be suspicious if a genre's average Global_Sales fell outside its observed minimum and maximum sales values, if negative sales values appeared, or if the highest average was based on only a very small number of observations. I checked the count, minimum, maximum, and mean Global_Sales for each genre. Platform has 888 observations with sales ranging from 0.01 to 40.24 million units and an average of approximately 0.93 million units. Although I originally predicted Shooter would have the highest average, Platform's result is supported by a substantial number of observations and its average falls within its observed range. Therefore, the result appears reasonable in the context of the dataset, and I did not change the result to match my original prediction.

### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer:


Platform games had the highest average global sales at approximately 0.93 million units per game, compared with approximately 0.80 million units for Shooter games. This did not support my original prediction that Shooter games would have the highest average global sales. Instead, the analysis showed that Platform games had the highest average in this dataset.

### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer:


A reader should not conclude that every Platform game sells more copies than games in other genres or that being a Platform game causes higher sales. This analysis compares average global sales, and individual game sales vary substantially within each genre. For example, Platform game sales in this dataset range from 0.01 million to 40.24 million units. The results are also limited to the games included in this dataset and may not represent video games released outside the dataset's coverage.

## Investigation 2

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 2?**

Your answer:


How do average North American sales compare with average European sales across video game genres?

### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer:


I predict that most video game genres will have higher average sales in North America than in Europe. I also expect the size of the difference between North American and European sales to vary by genre.

### Step 2: Data


Reuse the DataFrame and initial inspection from Investigation 1. Identify the observations and features relevant to this question; inspect additional features only if needed.


**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer:


I will use the Genre, NA_Sales, and EU_Sales features. Genre identifies the category of each video game, while NA_Sales and EU_Sales represent sales in North America and Europe, measured in millions of units. I will compare the average North American and European sales within each genre. Observations missing any of these required values will be excluded from the comparison so that both regional averages are based on usable data. Since this investigation compares averages rather than percentages, a denominator does not need to be defined.

### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [12]:
regional_sales_df = video_games_df.dropna(
    subset=["Genre", "NA_Sales", "EU_Sales"]
)

regional_sales_df[["Genre", "NA_Sales", "EU_Sales"]].head()


,Genre,NA_Sales,EU_Sales
0,Sports,41.36,28.96
1,Platform,29.08,3.58
2,Racing,15.68,12.76
3,Sports,15.61,10.93
4,Role-Playing,11.27,8.89


In [14]:
regional_sales_df.groupby("Genre")[["NA_Sales", "EU_Sales"]].mean()

,NA_Sales,EU_Sales
Genre,,
Action,0.260834,0.154045
Adventure,0.080783,0.048764
Fighting,0.263086,0.118174
Misc,0.232726,0.121566
Platform,0.501689,0.225619
Puzzle,0.211845,0.086224
Racing,0.287710,0.189359
Role-Playing,0.220540,0.125807
Shooter,0.447649,0.239864


**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer:


Average North American sales were higher than average European sales for every video game genre in the dataset. For example, Platform games averaged approximately 0.50 million units in North America compared with 0.23 million units in Europe, while Shooter games averaged approximately 0.45 million units in North America compared with 0.24 million units in Europe. Sports games averaged approximately 0.29 million units in North America compared with 0.16 million units in Europe. I calculated these results by grouping the observations by Genre and calculating the mean NA_Sales and EU_Sales for each group. This allows the two regional averages to be compared within each genre.

### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [15]:
regional_means_df = regional_sales_df.groupby("Genre")[["NA_Sales", "EU_Sales"]].mean()
regional_means_df["Difference"] = regional_means_df["NA_Sales"] - regional_means_df["EU_Sales"]
regional_means_df

,NA_Sales,EU_Sales,Difference
Genre,,,
Action,0.260834,0.154045,0.106789
Adventure,0.080783,0.048764,0.032018
Fighting,0.263086,0.118174,0.144912
Misc,0.232726,0.121566,0.111160
Platform,0.501689,0.225619,0.276070
Puzzle,0.211845,0.086224,0.125621
Racing,0.287710,0.189359,0.098351
Role-Playing,0.220540,0.125807,0.094733
Shooter,0.447649,0.239864,0.207785


**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer:


I would be suspicious if the calculated differences did not match the North American and European averages or if the sales values were negative. I calculated the difference between average NA_Sales and EU_Sales for each genre. Every difference was positive, confirming that every genre in the dataset had higher average sales in North America than in Europe. For example, Platform games had the largest difference at approximately 0.276 million units, while Adventure games had a smaller difference of approximately 0.032 million units. The results are consistent with the original sales averages, so the evidence makes sense in context.

### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer:


The results show that every video game genre had higher average sales in North America than in Europe. For example, Platform games averaged approximately 0.502 million units in North America compared with 0.226 million units in Europe, a difference of approximately 0.276 million units. Shooter games averaged approximately 0.448 million units in North America compared with 0.240 million units in Europe. This supports my prediction that average North American sales would generally be higher than European sales across genres.

### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer:


A reader should not conclude that every individual video game sells better in North America than in Europe. This analysis compares average sales within each genre, so individual games may have different sales patterns. The dataset also represents games and sales recorded through its collection period and may not represent current video game sales or future regional preferences.

## Investigation 3

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 3?**

Your answer:


What percentage of video games in the dataset received a critic score of 80 or higher, and how does that percentage differ among the major video game genres?

### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer:


I predict that less than half of the games in the dataset with recorded critic scores will have a Critic_Score of 80 or higher. I also predict that the percentage will vary by genre because some genres may contain a larger proportion of highly rated games than others.

### Step 2: Data


Reuse the DataFrame and initial inspection from Investigation 1. Identify the observations and features relevant to this question; inspect additional features only if needed.


**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer:


I will use the Genre and Critic_Score features. Critic_Score represents the game's critic rating on a scale from 0 to 100. I will remove observations that are missing either Genre or Critic_Score because they cannot be used to determine whether a game received a score of 80 or higher or to compare percentages among genres. For each genre, the denominator will be the total number of games in that genre that have a recorded Critic_Score. The numerator will be the number of those games with a Critic_Score of 80 or higher. The percentage will be calculated by dividing the numerator by the denominator and multiplying by 100.

### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [16]:
critic_genre_df = video_games_df.dropna(
    subset=["Genre", "Critic_Score"]
)

critic_genre_df[["Genre", "Critic_Score"]].head()

,Genre,Critic_Score
0,Sports,76.0
2,Racing,82.0
3,Sports,80.0
6,Platform,89.0
7,Misc,58.0


In [18]:
critic_genre_df = critic_genre_df.copy()

critic_genre_df["Score_80_Plus"] = critic_genre_df["Critic_Score"] >= 80

critic_genre_df[["Genre", "Critic_Score", "Score_80_Plus"]].head()


,Genre,Critic_Score,Score_80_Plus
0,Sports,76.0,False
2,Racing,82.0,True
3,Sports,80.0,True
6,Platform,89.0,True
7,Misc,58.0,False


In [19]:
critic_genre_df.groupby("Genre")["Score_80_Plus"].mean().mul(100).sort_values(ascending=False)

Genre
Sports          35.092127
Role-Playing    31.614654
Shooter         29.237288
Strategy        28.145695
Fighting        26.161369
Racing          24.932615
Platform        20.120724
Simulation      19.886364
Action          19.206349
Misc            16.826004
Adventure       16.099071
Puzzle          15.178571
Name: Score_80_Plus, dtype: float64

**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer:


I created a Boolean feature called Score_80_Plus that identifies whether each game with a recorded critic score received a Critic_Score of 80 or higher. I then grouped the games by Genre and calculated the percentage of True values within each genre. Sports had the highest percentage of games scoring 80 or higher at approximately 35.09%, followed by Role-Playing at 31.61% and Shooter at 29.24%. Puzzle had the lowest percentage at approximately 15.18%. These calculations show that the percentage of games receiving a critic score of 80 or higher varies considerably among genres.

### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [20]:
critic_genre_df.groupby("Score_80_Plus")["Critic_Score"].agg(["count", "min", "max"])

,count,min,max
Score_80_Plus,,,
False,6125,13.0,79.0
True,2012,80.0,98.0


**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer:


I would be suspicious if the Score_80_Plus indicator included Critic_Score values below 80 in the True group or values of 80 or higher in the False group. The sanity check shows that the False group ranges from 13 to 79, while the True group ranges from 80 to 98. This confirms that the calculated indicator correctly identifies games with critic scores of at least 80. The evidence therefore makes sense in context, and I did not find an issue that required correction.

### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer:


Sports games had the highest percentage of critic scores of 80 or higher at approximately 35.09%. Role-Playing games were second at approximately 31.61%, followed by Shooter games at approximately 29.24%. This [supported/did not support] my prediction that ______ would have the highest percentage of games with critic scores of at least 80.

### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer:


A reader should not conclude that Sports games are automatically better than games in other genres. This analysis only compares the percentage of games in each genre that received a Critic_Score of 80 or higher among games with available critic scores. It does not account for differences in the number of reviewed games, individual reviewer opinions, release year, platform, or other factors that may affect critic scores.

## AI-use reflection


**State whether you used an AI tool. If you did, describe one suggestion you checked and what you accepted, changed, or rejected. If you did not, state that you did not use one.**

Your answer:


I used ChatGPT as an AI tool during this assignment. One suggestion I checked was using the Pandas groupby() function to organize the video game data by genre. I ran the suggested code in my notebook and reviewed the output to make sure the calculations matched the data. After verifying that the results were reasonable, I accepted the suggestion and used the grouped results in my analysis.

## Submission checklist

- Include your chosen dataset, source, and dictionary reference.
- Complete three distinct questions, each with a prediction and all six workflow steps.
- Use a descriptive DataFrame name ending in `_df` and relative file paths.
- Restart the kernel and run all cells in order; verify that numerical claims match displayed output.
- Include a contextual sanity check, supported conclusion, and relevant limitation for each question.
- Complete every written response and the AI-use reflection.
